Realm/Realm/DatabaseTypes/RealmCollectionBase.cs

```diff
@@ -488,8 +488,16 @@ internal Enumerator(RealmCollectionBase<T> parent)
 
                 // If we didn't snapshot the parent, we should not dispose the results handle, otherwise we'll invalidate the
                 // parent collection after iterating it. Only collections of objects support snapshotting.
-                _shouldDisposeHandle = parent.IsValid && parent.Handle.Value.CanSnapshot && parent.Metadata != null;
-                _enumerating = _shouldDisposeHandle ? new RealmResults<T>(parent.Realm, parent.Handle.Value.Snapshot(), parent.Metadata) : parent;
+                _shouldDisposeHandle = parent.IsValid && !parent.IsFrozen && parent.Handle.Value.CanSnapshot && parent.Metadata != null;
+
+                if (parent.IsFrozen)
```

Comment:
Do we need this check here? Since `_shouldDisposeHandle = ... && !parent.IsFrozen`, I would expect `_shouldDisposeHandle` to be `false` for frozen collections, which means that the ternary on line 499 would have been sufficient.

Reply:
that's a great point, I will remove this and perhaps add a comment specifying the isFrozen bit